gunicorn/http/wsgi.py

```diff
@@ -234,7 +233,7 @@ def start_response(self, status, headers, exc_info=None):
         if exc_info:
             try:
                 if self.status and self.headers_sent:
-                    reraise(exc_info[0], exc_info[1], exc_info[2])
+                    raise exc_info
```

Comment:
The wsgiref module does this:

```py
        if exc_info:
            try:
                if self.headers_sent:
                    # Re-raise original exception if headers sent
                    raise exc_info[0](exc_info[1]).with_traceback(exc_info[2])
            finally:
                exc_info = None        # avoid dangling circular ref
```

https://github.com/python/cpython/blob/master/Lib/wsgiref/handlers.py#L215-L221

six's implementation is more defensive:

https://github.com/benjaminp/six/blob/master/six.py#L687-L696

We can copy six's implementation into `gunicorn/util.py`.

Reply:
And note that `reraise()` from six ignores `exc_type`. This would cause a problem if we want to raise the exception with a different exception type, but since we no longer support Python 2, we can do the same thing with `raise new from old`.

Reply:
✅